### PR TITLE
Fixed #2075 - fixes validation checkmarks

### LINKED ISSUE #2075: Missing green checkmarks when adding URL and description
In the existing webcompat form, when you put in a URL and description, you get a green checkmark.

In the form on staging, nothing happens there. Seems like a bug.

### PR DESCRIPTION
**Pull request by zoepage**
r? @miketaylr

### CODE REVIEW

File: webcompat/static/js/lib/bugform.js [hunk 1]
```diff
@@ -292,11 +297,13 @@ function BugForm() {
 
   /* Check if Browser and OS are empty or not, only
      so we can set them to valid (there is no invalid state) */
-  this.checkOptionalNonEmpty = function() {
+  this.checkOptionalNonEmpty = function(types) {
     _.forEach(
-      [this.browserField, this.osField],
-      _.bind(function(input) {
+      [...types],
```
Comment (miketaylr) [inline webcompat/static/js/lib/bugform.js]: We can't use es6 rest because it will break the reporting form for IE users, and other older browsers. :(

File: webcompat/static/js/lib/bugform.js [hunk 2]
```diff
@@ -292,11 +297,13 @@ function BugForm() {
...
+      _.bind(function(key) {
+        const input = this[key];
```
Comment (miketaylr) [inline webcompat/static/js/lib/bugform.js]: Can we change this to `var` please? We're doing ES5 until we decide we're ready to ship ES6 (or transpile -- but that's probably a future discussion).

File: webcompat/static/js/lib/bugform.js [hunk 3]
```diff
@@ -77,9 +77,14 @@ function BugForm() {
...
-    this.osField
-      .add(this.browserField)
-      .on("blur input", _.bind(this.checkOptionalNonEmpty, this));
+    this.osField.on(
+      "blur",
+      _.bind(this.checkOptionalNonEmpty, this, ["osField"])
+    );
+    this.browserField.on(
+      "blur",
+      _.bind(this.checkOptionalNonEmpty, this, ["browserField"])
```
Comment (miketaylr) [inline webcompat/static/js/lib/bugform.js]: Any reason to not just pass in the reference to the DOM elements directory here?

File: webcompat/static/js/lib/bugform.js (hunk 1 above)
Comment (zoepage) [reply webcompat/static/js/lib/bugform.js]: Sorry, I shouldn't switch between projects. 😅
Will send a fix.

File: webcompat/static/js/lib/bugform.js (hunk 3 above)
Comment (zoepage) [reply webcompat/static/js/lib/bugform.js]: I'm not sure I understand the wording. 🤔

Comment (miketaylr) [reply webcompat/static/js/lib/bugform.js]: Let me rephrase:

You're changing `checkOptionalNonEmpty` to accept an argument `types`, which is an array of keys. For each of those keys, you grab a reference to the corresponding DOM input element via `input = this[key];`.

Why not just pass in `this.browserField` directly and skip passing the array of strings that ultimately just returns `this.browserField`?

`_.bind(this.checkOptionalNonEmpty, this, [this.browserField])`, etc.

Comment (zoepage) [reply webcompat/static/js/lib/bugform.js]: This is actually exactly what I did today. Small fix and will update the PR :)

File: webcompat/static/js/lib/bugform.js [hunk 4]
```diff
@@ -77,9 +77,14 @@ function BugForm() {
...
-    this.osField
-      .add(this.browserField)
-      .on("blur input", _.bind(this.checkOptionalNonEmpty, this));
+    this.osField.on(
+      "blur",
+      _.bind(this.checkOptionalNonEmpty, this, this.osField)
+    );
+    this.browserField.on(
+      "blur",
+      _.bind(this.checkOptionalNonEmpty, this, this.browserField)
```
Comment (miketaylr) [inline webcompat/static/js/lib/bugform.js]: If we're going to do it this way, we don't even need to pass in `this.browserField` -- we can just get that from `event.target` down in `checkOptionalNonEmpty`, right?

File: webcompat/static/js/lib/bugform.js [hunk 5]
```diff
@@ -77,9 +77,14 @@ function BugForm() {
     );
     this.problemType.on("change", _.bind(this.checkProblemTypeValidity, this));
     this.uploadField.on("change", _.bind(this.checkImageTypeValidity, this));
-    this.osField
-      .add(this.browserField)
-      .on("blur input", _.bind(this.checkOptionalNonEmpty, this));
+    this.osField.on(
+      "blur",
```
Comment (miketaylr) [inline webcompat/static/js/lib/bugform.js]: Is removing the `input` handler intentional?

**Comment by miketaylr** [comment]
> And if we run through the inputs with the if, we get rid of the for loop. Felt bit more efficient.

(I'm not sure looping over a list of 2 items is a significant perf bottleneck here. :))

File: webcompat/static/js/lib/bugform.js (hunk 5 above)
Comment (zoepage) [reply webcompat/static/js/lib/bugform.js]: adding both would mean to add a bind to both. I don't feel this is needed. Am I missing something?

File: webcompat/static/js/lib/bugform.js (hunk 4 above)
Comment (zoepage) [reply webcompat/static/js/lib/bugform.js]: With `this.browserField` we have already set a reference and I feel like this is more readable for someone not knowing the code, but maybe I am missing something?

Comment (miketaylr) [reply webcompat/static/js/lib/bugform.js]: Sure, I guess it's just a question of style preference.

File: webcompat/static/js/lib/bugform.js (hunk 5 above)
Comment (miketaylr) [reply webcompat/static/js/lib/bugform.js]: Not sure I follow, let me try again:

Before we had `$(foo).on("blur input", bar)`. Now we have $(foo).on("blur", bar)`, meaning we're not handling the `input` event.

What do you mean by adding `bind` to both? I don't follow. Thanks!

Comment (zoepage) [reply webcompat/static/js/lib/bugform.js]: Sorry for the short and confusing answer. Okay, let me rephrase it. :)

So, we add `blur` and `input` as events to trigger the function. I think, `blur` is enough as this will add / remove the checkmark as needed and we don't fire the function on every change of the input value. So the answer to your question `Is removing the input handler intentional?` is yes. 

If you think, this feels weird or does not cover all use cases, I am ok to put it back. I guess, this is mostly about UX preference.

Comment (miketaylr) [reply webcompat/static/js/lib/bugform.js]: SGTM. Let's merge now and if it feels wrong later on, file a bug to revisit/change it back.
